src/main/java/org/apache/sling/scripting/sightly/render/ObjectModel.java

```diff
@@ -308,12 +306,7 @@ public static String toString(Object object) {
             Iterable<Object> iterable = (Iterable<Object>) object;
             return fromIterator(iterable.iterator());
         }
-        if (object instanceof String || object instanceof Number) {
-            Collection<Object> list = new ArrayList<>();
-            list.add(object);
-            return list;
-        }
-        return Collections.emptyList();
+        return Collections.singletonList(object);
```

Comment:
While I agree with the change, I think it comes with a certain risk as it changes the returned collection from a modifiable to an unmodifiable one. Should we make that somehow clear to the consumers? 

In general I think we should return unmodifiable collections. Can you change `fromIterator()` accordingly as well?

Reply:
The empty list being returned before was also immutable.

Reply:
But not the `ArrayList` returned with the single String | Number.

Reply:
Modifying the other collections to return immutable lists consistently requiress quite some updates in the test (which currently compare with equals against a mutable list), therefore I would like to separate this from this extension. Changing the single element list to be immutable didn't require any test changes.

Reply:
After having a second thought it wasn't that much effort and only one change in the unit test was necessary, therefore I committed those changes in https://github.com/apache/sling-org-apache-sling-scripting-sightly-runtime/pull/9/commits/a16076eca5099ef4134f5fa90adb76a6af624df4.